src/run.ts

```diff
@@ -1,147 +1,99 @@
-import * as toolCache from '@actions/tool-cache';
 import * as core from '@actions/core';
-import { ToolRunner, argStringToArray } from "@actions/exec/lib/toolrunner";
 
-import * as path from 'path';
-import * as os from 'os';
-import * as io from '@actions/io';
+const k8s = require('@kubernetes/client-node');
 
-import fileUtility = require('./file.utility')
+import { CoreV1Api, KubeConfig, V1ObjectMeta, V1Secret } from '@kubernetes/client-node';
 
-let kubectlPath = "";
-
-async function checkAndSetKubectlPath() {
-    kubectlPath = await io.which('kubectl', false);
-    if (kubectlPath) {
-        return;
-    }
-
-    const allVersions = toolCache.findAllVersions('kubectl');
-    kubectlPath = allVersions.length > 0 ? toolCache.find('kubectl', allVersions[0]) : '';
-    if (!kubectlPath) {
-        throw new Error('Kubectl is not installed');
-    }
-
-    kubectlPath = path.join(kubectlPath, `kubectl${getExecutableExtension()}`);
-}
-
-function getExecutableExtension(): string {
-    if (os.type().match(/^Win/)) {
-        return '.exe';
+export function checkClusterContext() {
+    if (!process.env["KUBECONFIG"]) {
+        throw new Error('Cluster context not set. Use k8s-set-context/aks-set-context action to set cluster context');
     }
-
-    return '';
 }
 
-async function createSecret() {
-    const typeOfSecret = core.getInput('secret-type', { required: true });
-    const secretName = core.getInput('secret-name', { required: true });
-    const namespace = core.getInput('namespace');
-
-    await deleteSecret(namespace, secretName);
+export async function buildSecret(secretName: string, namespace: string): Promise<V1Secret> {
+    // The secret type for the new secret
+    const secretType: string = core.getInput('secret-type', { required: true });
 
-    let args;
-    if (typeOfSecret === "docker-registry") {
-        args = getDockerSecretArguments(secretName);
+    let metaData: V1ObjectMeta = {
+        name: secretName,
+        namespace: namespace
     }
-    else if (typeOfSecret === "generic") {
-        args = getGenericSecretArguments(secretName);
-    }
-    else {
-        throw new Error('Invalid secret-type input. It should be either docker-registry or generic');
+
+    // The serialized form of the secret data is a base64 encoded string
+    let data: { [key: string]: string } = {}
+    if (core.getInput('data')) {
```

Comment:
this if can just be remove with using the {required: true} optional parameter

Reply:
since they are optional fields, I switched to using core.debug so the intent of the print statement is clearer in that it is only for debugging purposes